components/esmloader/load.cpp

```diff
@@ -215,6 +215,8 @@ namespace EsmLoader
                 reader.setEncoder(encoder);
                 reader.setIndex(static_cast<int>(i));
                 reader.open(collection.getPath(file).string());
+                if (query.mLoadCells)
+                    reader.resolveParentFileIndices(readers);
```

Comment:
I'm not sure this call produces expected result.

For example I used this list of groundcover mods:
```
groundcover=Grass Bloodmoon.esp
groundcover=Grass Vanilla 1.esp
groundcover=Grass Vanilla 2.esp
groundcover=Grass TR AC.esp
groundcover=Grass TR AI.esp
groundcover=Grass TR AL.esp
groundcover=Grass TR AT.esp
groundcover=Grass TR BC.esp
groundcover=Grass TR GL.esp
groundcover=Grass TR SV.esp
groundcover=Grass TR WG.esp
groundcover=Cyrodiil_Grass.ESP
groundcover=Sky_Main_Grass.ESP
groundcover=Grass rebirth 1.esp
groundcover=Grass rebirth 2.esp
```

First 11 esp are from Aesthesia groundcover, last 2 are from Grass Rebirth 5.4 mentioned in [this](https://github.com/OpenMW/openmw/pull/3206#issuecomment-955142747) comment.

This call adds indices only for `Cyrodiil_Grass.ESP` and `Sky_Main_Grass.ESP`. For `Cyrodiil_Grass.ESP` it adds 3 same numbers into the `mCtx.parentFileIndices`. In my case it's 11 (the index of `Cyrodiil_Grass.ESP` in the list), so `parentFileIndices` stores `{11, 11, 11}` after this call.

`Cyrodiil_Grass.ESP` lists `Morrowind.esm`, `Tribunal.esm` and `Bloodmoon.esm` as master files in the header but they are not present in the list when `EsmLoader::loadEsmData` is called only for groundcover files. This doesn't look right.

Reply:
Well, both in master and here "content" game files do not present in groundcover store, so their indices obviously can not be resolved. Missing dependencies seem to be resolved to the plugin itself as a fallback. So instances which refer to this file are treated as added by plugin itself.

For "content" entries such situation does not happen because there is a validation which does not allow to lauch the game with unresolved dependencies. As for "groundcover" files, often they do not have dependent master files at all (what seems to be a case for a part of your files).  The whole purpose of indices here is to handle a quite rare case when one "groundcover" file adds grass instances while an another dependent "groundcover" file moves or removes them.

In theory, we can resolve content file index to some kind of special value inside `resolveParentFileIndices` instead (to -1, for example) and skip instances with such index when we build groundcover chunks. It should provide us a more reliable way to ignore data from "content" files, but I am not sure if it worth it.

Reply:
Ok, I think it makes sense to leave as is. I wonder why a mod has to depend on multiple master files when it can be ignored, but it's a question for mod-makers.